controllers/statusreport/statusreport_adapter.go

```diff
@@ -81,11 +81,6 @@ func (a *Adapter) EnsureSnapshotTestStatusReportedToGitProvider() (controller.Op
 	if err != nil {
 		a.logger.Error(err, "failed to report test status to git provider for snapshot",
 			"snapshot.Namespace", a.snapshot.Namespace, "snapshot.Name", a.snapshot.Name)
-		if helpers.IsObjectYoungerThanThreshold(a.snapshot, SnapshotRetryTimeout) {
-			return controller.RequeueWithError(err)
-		} else {
-			return controller.ContinueProcessing()
```

Comment:
Should we just skip this line and leave the `helpers.IsObjectYoungerThanThreshold()` call?

Reply:
I don't follow what you mean

Reply:
I mean that we could keep the logic to [requeue if we encounter an error within the threshold](https://github.com/redhat-appstudio/integration-service/pull/606/files#diff-d00409fb87121cfd0dfb79643900ddc48645e414a6029ded78c2a0f50b3d1c41L84-L85), but remove the call to `ContinueProcessing()` here and instead proceed to remove the finalizers in the lower lines.

Reply:
I guess Kruno means keeping
```
if helpers.IsObjectYoungerThanThreshold(a.snapshot, SnapshotRetryTimeout) {
	return controller.RequeueWithError(err)
}
```
Right? I just guess.

Then if it still fails after several tries, then we remove the finalizer from PLR directly.

Reply:
it would block finalizers for 3 hours (for nothing)

Reply:
before bc75896 it was not retried at all, just well hidden in CRA creation

Reply:
My vote then is to also reduce the [SnapshotRetryTimeout](https://github.com/redhat-appstudio/integration-service/blob/1ab32ad73a03c81a0395ad67c317bf5e53ee512e/controllers/statusreport/statusreport_adapter.go#L39) to ~15mins.

Reply:
this creation time is not the greatest, for long running tests it may not retry even once

Reply:
I added new commit how to address this issue to avoid retries on something that retries won't help, please check, if you like it I can rework it

Reply:
I like the new solution, does that mean that the changes in this file don't need to happen?

Reply:
Right, fully agree on that.